### PR TITLE
Fix email notifications for shows with special cars.

### LINKED ISSUE #9500: Special characters break show notification emails
**Describe the bug**
Emails are no longer being sent to recipients in the "Show notification list" if the show name contains special characters. For example, Mayans M.C, Rebel (2021), Restaurant: Impossible. All shows without special characters still work fine.

**To Reproduce**
1. Add an email to the "Show notification list" for a show with special characters in its title.
2. Allow show to be processed
3. No email sent

**Expected behavior**
Email should be sent to recipient listed.

**Screenshots**
The "Show notification list" is located in Settings -> Notifications -> Social -> Email
70375687-5a8cf580-18c6-11ea-871a-6918240fcf55

**Medusa (please complete the following information):**
 - OS: Unraid 6.9.2 (binhex-medusa docker)
 - Branch: master
 - Commit: Unknown
 - Python version: 3.9.2 (default, Feb 20 2021, 18:40:11) [GCC 10.2.0]
 - Database version: 44.18

**Debug logs (at least 50 lines):**
General > Advanced Settings > Enable debug
<details>

```
2021-05-05 00:10:01 DEBUG    TORNADO :: [] API :: 192.168.**.** - gave correct API KEY. ACCESS GRANTED
2021-05-05 00:10:01 DEBUG    TORNADO :: [] API :: all args: ()
2021-05-05 00:10:01 DEBUG    TORNADO :: [] API :: all kwargs: "{'cmd': 'postprocess', 'path': '/media/Downloads/deluge/tv', 'force_replace': '0', 'return_data': '0', 'process_method': 'move', 'is_priority': '0', 'delete_files': '1', 'failed': '0', 'type': 'auto', 'jsonp': 'foo'}"
2021-05-05 00:10:01 DEBUG    TORNADO :: [] API :: postprocess: {'path': '/media/Downloads/deluge/tv', 'force_replace': '0', 'return_data': '0', 'process_method': 'move', 'is_priority': '0', 'delete_files': '1', 'failed': '0', 'type': 'auto', 'jsonp': 'foo'}
2021-05-05 00:10:01 INFO     TORNADO :: [] Processing path: /media/Downloads/deluge/tv
2021-05-05 00:10:01 DEBUG    TORNADO :: [] No resource_name passed, using path [/media/Downloads/deluge/tv] to process as a source folder
2021-05-05 00:10:01 DEBUG    TORNADO :: [] Processing folder: /media/Downloads/deluge/tv
2021-05-05 00:10:01 DEBUG    TORNADO :: [] Post-processing files: ['Mayans.M.C.S03E09.480p.x264-mSD[eztv.re].mkv']
2021-05-05 00:10:01 DEBUG    TORNADO :: [] Post-processing video files: ['Mayans.M.C.S03E09.480p.x264-mSD[eztv.re].mkv']
2021-05-05 00:10:01 INFO     TORNADO :: [] Processing /media/Downloads/deluge/tv/Mayans.M.C.S03E09.480p.x264-mSD[eztv.re].mkv
2021-05-05 00:10:01 DEBUG    TORNADO :: [] Analyzing name: Mayans.M.C.S03E09.480p.x264-mSD[eztv.re].mkv
2021-05-05 00:10:01 DEBUG    TORNADO :: [] Using guessit cache item for Mayans.M.C.S03E09.480p.x264-mSD[eztv.re].mkv
2021-05-05 00:10:01 DEBUG    TORNADO :: [] Found numbering S03E09 from parser for show Mayans M.C. S03E09
2021-05-05 00:10:01 DEBUG    TORNADO :: [] Converted parsed result Mayans.M.C.S03E09.480p.x264-mSD[eztv.re].mkv into title: Mayans M.C., season: 3, episode: [9], screen_size: 480p, video_codec: H.264, video_encoder: x264, release_group: mSD, container: mkv, type: episode, parsing_time: 0.043365478515625, absolute_episode: [], quality: SDTV, total_time: None
2021-05-05 00:10:01 DEBUG    TORNADO :: [] Matched release Mayans.M.C.S03E09.480p.x264-mSD[eztv.re].mkv to a series in your database: Mayans M.C.
2021-05-05 00:10:01 DEBUG    TORNADO :: [] Parsed Mayans.M.C.S03E09.480p.x264-mSD[eztv.re].mkv into title: Mayans M.C., season: 3, episode: [9], screen_size: 480p, video_codec: H.264, video_encoder: x264, release_group: mSD, container: mkv, type: episode, parsing_time: 0.043365478515625, absolute_episode: [], quality: SDTV, total_time: 0.044821977615356445
2021-05-05 00:10:01 DEBUG    TORNADO :: [] Show information parsed from file name
2021-05-05 00:10:01 DEBUG    TORNADO :: [] Retrieving episode object for Mayans M.C. S03E09
2021-05-05 00:10:01 DEBUG    TORNADO :: [] The episode file has a quality in it, using that: SDTV
2021-05-05 00:10:01 DEBUG    TORNADO :: [] This episode was found in history as SNATCHED.
2021-05-05 00:10:01 DEBUG    TORNADO :: [] Quality of the episode we are processing: SDTV
2021-05-05 00:10:01 DEBUG    TORNADO :: [] Snatch in history: True
2021-05-05 00:10:01 DEBUG    TORNADO :: [] Manually snatched: False
2021-05-05 00:10:01 DEBUG    TORNADO :: [] Info hash: 3c4f6bf04416df171026999464720eb38602e373
2021-05-05 00:10:01 DEBUG    TORNADO :: [] NZB: False
2021-05-05 00:10:01 DEBUG    TORNADO :: [] Current quality: SDTV
2021-05-05 00:10:01 DEBUG    TORNADO :: [] New quality: SDTV
2021-05-05 00:10:01 DEBUG    TORNADO :: [] Proper: False
2021-05-05 00:10:01 DEBUG    TORNADO :: [] This episode is a priority download: True
2021-05-05 00:10:01 INFO     TORNADO :: [] New file: /media/Downloads/deluge/tv/Mayans.M.C.S03E09.480p.x264-mSD[eztv.re].mkv
2021-05-05 00:10:01 INFO     TORNADO :: [] New size: 428.01 MB
2021-05-05 00:10:01 DEBUG    TORNADO :: [] There is no existing file so there's no worries about replacing it
2021-05-05 00:10:01 INFO     TORNADO :: [] This download is marked a priority download so I'm going to replace an existing file if I find one
2021-05-05 00:10:01 DEBUG    TORNADO :: [] Trying to determine free space on destination drive
2021-05-05 00:10:01 DEBUG    TORNADO :: [] Found release name Mayans.M.C.S03E09.480p.x264-mSD[eztv.re]
2021-05-05 00:10:01 DEBUG    TORNADO :: [] Using guessit cache item for tv
2021-05-05 00:10:01 DEBUG    TORNADO :: [] Using cache item for tv
2021-05-05 00:10:01 DEBUG    TORNADO :: [] Folder name (tv) appears to be a valid release name. Using it.
2021-05-05 00:10:01 DEBUG    TORNADO :: [] Using guessit cache item for Mayans.M.C.S03E09.480p.x264-mSD[eztv.re]
2021-05-05 00:10:01 DEBUG    TORNADO :: [] Matched release Mayans.M.C.S03E09.480p.x264-mSD[eztv.re] to a series in your database: Mayans M.C.
2021-05-05 00:10:01 DEBUG    TORNADO :: [] Parsed Mayans.M.C.S03E09.480p.x264-mSD[eztv.re] into title: Mayans M.C., season: 3, episode: [9], screen_size: 480p, video_codec: H.264, video_encoder: x264, release_group: mSD, type: episode, parsing_time: 0.05014157295227051, absolute_episode: [], quality: SDTV, total_time: 0.050296783447265625
2021-05-05 00:10:01 DEBUG    TORNADO :: [] Found codec for Mayans M.C. The House of Death Floats By
2021-05-05 00:10:01 DEBUG    TORNADO :: [] 349659: Formatting pattern: %SN - %0Sx09 - %EN -> Mayans M.C - 03x09 - The House of Death Floats By
2021-05-05 00:10:01 DEBUG    TORNADO :: [] Using guessit cache item for Mayans.M.C.S03E09.480p.x264-mSD[eztv.re]
2021-05-05 00:10:01 DEBUG    TORNADO :: [] Using cache item for Mayans.M.C.S03E09.480p.x264-mSD[eztv.re]
2021-05-05 00:10:01 DEBUG    TORNADO :: [] Matched release Mayans.M.C.S03E09.480p.x264-mSD[eztv.re] to a series in your database: Mayans M.C.
2021-05-05 00:10:01 DEBUG    TORNADO :: [] Parsed Mayans.M.C.S03E09.480p.x264-mSD[eztv.re] into title: Mayans M.C., season: 3, episode: [9], screen_size: 480p, video_codec: H.264, video_encoder: x264, release_group: mSD, type: episode, parsing_time: 0.002591848373413086, absolute_episode: [], quality: SDTV, total_time: 0.0027403831481933594
2021-05-05 00:10:01 DEBUG    TORNADO :: [] Found codec for Mayans M.C. The House of Death Floats By
2021-05-05 00:10:01 DEBUG    TORNADO :: [] 349659: Formatting pattern: Season %0S -> Season 03
2021-05-05 00:10:01 DEBUG    TORNADO :: [] Destination folder for this episode: /media/TV/Mayans M.C/Season 03
2021-05-05 00:10:01 DEBUG    TORNADO :: [] Checking if the path /media/TV/Mayans M.C/Season 03 already exists
2021-05-05 00:10:01 DEBUG    TORNADO :: [] No associated files were found for /media/Downloads/deluge/tv/Mayans.M.C.S03E09.480p.x264-mSD[eztv.re].mkv
2021-05-05 00:10:01 DEBUG    TORNADO :: [] Moving file from /media/Downloads/deluge/tv/Mayans.M.C.S03E09.480p.x264-mSD[eztv.re].mkv to /media/TV/Mayans M.C/Season 03/Mayans M.C - 03x09 - The House of Death Floats By.mkv 
2021-05-05 00:10:01 DEBUG    TORNADO :: [] 349659: Setter sets location to /media/TV/Mayans M.C/Season 03/Mayans M.C - 03x09 - The House of Death Floats By.mkv
2021-05-05 00:10:01 DEBUG    TORNADO :: [] Transaction with 1 queries executed
2021-05-05 00:10:01 DEBUG    TORNADO :: [] Notification for NMJ scan update not enabled, skipping this notification
2021-05-05 00:10:01 DEBUG    TORNADO :: [] Using guessit cache item for /media/TV/Mayans M.C/Season 03/Mayans M.C - 03x09 - The House of Death Floats By
2021-05-05 00:10:01 DEBUG    TORNADO :: [] Matched release /media/TV/Mayans M.C/Season 03/Mayans M.C - 03x09 - The House of Death Floats By to a series in your database: Mayans M.C.
2021-05-05 00:10:01 DEBUG    TORNADO :: [] Parsed /media/TV/Mayans M.C/Season 03/Mayans M.C - 03x09 - The House of Death Floats By into title: Mayans M C, season: 3, episode: [9], episode_title: The House of Death Floats By, type: episode, parsing_time: 0.07815694808959961, absolute_episode: [], quality: Unknown, total_time: 0.07839155197143555
2021-05-05 00:10:01 DEBUG    TORNADO :: [] Using guessit cache item for Mayans.M.C.S03E09.480p.x264-mSD[eztv.re]
2021-05-05 00:10:01 DEBUG    TORNADO :: [] Using cache item for Mayans.M.C.S03E09.480p.x264-mSD[eztv.re]
2021-05-05 00:10:01 DEBUG    TORNADO :: [] Matched release Mayans.M.C.S03E09.480p.x264-mSD[eztv.re] to a series in your database: Mayans M.C.
2021-05-05 00:10:01 DEBUG    TORNADO :: [] Parsed Mayans.M.C.S03E09.480p.x264-mSD[eztv.re] into title: Mayans M.C., season: 3, episode: [9], screen_size: 480p, video_codec: H.264, video_encoder: x264, release_group: mSD, type: episode, parsing_time: 0.003516674041748047, absolute_episode: [], quality: SDTV, total_time: 0.003659963607788086
2021-05-05 00:10:01 DEBUG    TORNADO :: [] Found codec for Mayans M.C. The House of Death Floats By
2021-05-05 00:10:01 DEBUG    TORNADO :: [] 349659: Formatting pattern: %SN - %Sx09 - %EN - %QN -> Mayans M.C - 3x09 - The House of Death Floats By - SDTV
2021-05-05 00:10:01 DEBUG    TORNADO :: [] Using guessit cache item for /media/TV/Mayans M.C/Season 03/Mayans M.C - 03x09 - The House of Death Floats By
2021-05-05 00:10:01 DEBUG    TORNADO :: [] Using cache item for /media/TV/Mayans M.C/Season 03/Mayans M.C - 03x09 - The House of Death Floats By
2021-05-05 00:10:01 DEBUG    TORNADO :: [] Matched release /media/TV/Mayans M.C/Season 03/Mayans M.C - 03x09 - The House of Death Floats By to a series in your database: Mayans M.C.
2021-05-05 00:10:01 DEBUG    TORNADO :: [] Parsed /media/TV/Mayans M.C/Season 03/Mayans M.C - 03x09 - The House of Death Floats By into title: Mayans M C, season: 3, episode: [9], episode_title: The House of Death Floats By, type: episode, parsing_time: 0.0027256011962890625, absolute_episode: [], quality: Unknown, total_time: 0.0029702186584472656
2021-05-05 00:10:01 DEBUG    TORNADO :: [] Using guessit cache item for Mayans.M.C.S03E09.480p.x264-mSD[eztv.re]
2021-05-05 00:10:01 DEBUG    TORNADO :: [] Using cache item for Mayans.M.C.S03E09.480p.x264-mSD[eztv.re]
2021-05-05 00:10:01 DEBUG    TORNADO :: [] Matched release Mayans.M.C.S03E09.480p.x264-mSD[eztv.re] to a series in your database: Mayans M.C.
2021-05-05 00:10:01 DEBUG    TORNADO :: [] Parsed Mayans.M.C.S03E09.480p.x264-mSD[eztv.re] into title: Mayans M.C., season: 3, episode: [9], screen_size: 480p, video_codec: H.264, video_encoder: x264, release_group: mSD, type: episode, parsing_time: 0.0026199817657470703, absolute_episode: [], quality: SDTV, total_time: 0.0027832984924316406
2021-05-05 00:10:01 DEBUG    TORNADO :: [] Found codec for Mayans M.C. The House of Death Floats By
2021-05-05 00:10:01 DEBUG    TORNADO :: [] 349659: Formatting pattern: %SN - %Sx09 - %EN - %QN -> Mayans M.C - 3x09 - The House of Death Floats By - SDTV
2021-05-05 00:10:01 DEBUG    TORNADO :: [] Email notifier parsed "Mayans M.C - 3x09 - The House of Death Floats By - SDTV" into {'show': 'Mayans M.C', 'ep_id': '3x09', 'episode': 'The House of Death Floats By - SDTV'}
2021-05-05 00:10:01 DEBUG    TORNADO :: [] Notification recipients: {'*******@gmail.com'}
2021-05-05 00:10:01 DEBUG    TORNADO :: [] HOST: smtp.gmail.com; PORT: 587; FROM: Server<**********@****.net>, TLS: 1, USER: **********, PWD: **********, TO: {'*******@gmail.com'}
2021-05-05 00:10:02 DEBUG    TORNADO :: [] Sending initial EHLO command!
2021-05-05 00:10:02 DEBUG    TORNADO :: [] Sending STARTTLS command!
2021-05-05 00:10:02 DEBUG    TORNADO :: [] Sending LOGIN command!
2021-05-05 00:10:03 DEBUG    TORNADO :: [] Download notification sent to {'*******@gmail.com'} for Mayans M.C - 3x09 - The House of Death Floats By - SDTV
2021-05-05 00:10:03 INFO     TORNADO :: [] Processing succeeded for /media/Downloads/deluge/tv/Mayans.M.C.S03E09.480p.x264-mSD[eztv.re].mkv
2021-05-05 00:10:03 INFO     TORNADO :: [] Post-processing completed.
```
</details>

**Additional context**
I have debug logs from the time an episode of Mayans M.C was processed but the "Show notification list" is not included in any logs. Only entries related to the "Global email list" are shown. I know from previous experience that these two lists are handled differently within medusa. Also, this problem only started with a medusa update from about two or three months ago. Before that everything worked great.

### PR DESCRIPTION
**Pull request by p0psicles**
* Make sure to look up the show by id's, in stead of by name.

- [x] PR is based on the DEVELOP branch
- [x] Don't send big changes all at once. Split up big PRs into multiple smaller PRs that are easier to manage and review
- [x] Read the [contribution guide](https://github.com/pymedusa/Medusa/blob/master/.github/CONTRIBUTING.md)

fix #9500

### CODE REVIEW

File: medusa/notifiers/emailnotify.py [hunk 1]
```diff
@@ -65,15 +65,15 @@ def test_notify(self, host, port, smtp_from, use_tls, user, pwd, to):
         msg['Date'] = formatdate(localtime=True)
         return self._sendmail(host, port, smtp_from, use_tls, user, pwd, [to], msg, True)
 
-    def notify_snatch(self, title, message):
+    def notify_snatch(self, title, message, ep_obj):
```
Comment (medariox) [inline medusa/notifiers/emailnotify.py:68]: I think you forgot to make it a keyword argument.

Comment (p0psicles) [reply medusa/notifiers/emailnotify.py:68]: I kinda did this on purpose. I had it as keyword first. But it's always called with an ep_obj. So no need to make it default to None.

Comment (medariox) [reply medusa/notifiers/emailnotify.py:68]: But doesn't this defeat the purpose to use **kwargs for all other notifiers? We are forced to always add the `email` argument for all notifiers like this.

Comment (p0psicles) [reply medusa/notifiers/emailnotify.py:68]: No, the **kwargs param just prevents them from Errorring. As it's now its not causing any errors.

Comment (medariox) [reply medusa/notifiers/emailnotify.py:68]: The point is that it is pretty confusing right now. The method header should be equal for all and the arguments should be handled inside the function IMO.

**Pull request merged**